src/charm.py

```diff
@@ -640,65 +649,95 @@ def _configure_bessd_workload(self) -> None:
-                time.sleep(2)
-        raise TimeoutError("Timed out trying to run configuration for bess")
 
-    def _configure_bessd_for_dpdk(self) -> None:
-        """Configures bessd container for DPDK."""
-        dpdk = DPDK(
-            statefulset_name=self.model.app.name,
-            namespace=self._namespace,
-            dpdk_access_interface_resource_name=DPDK_ACCESS_INTERFACE_RESOURCE_NAME,
-            dpdk_core_interface_resource_name=DPDK_CORE_INTERFACE_RESOURCE_NAME,
-        )
-        if not dpdk.is_configured(container_name=self._bessd_container_name):
-            dpdk.configure(container_name=self._bessd_container_name)
+        while not self._is_bessd_grpc_service_ready():
+            if time.time() - initial_time > timeout:
+                raise TimeoutError("Timed out waiting for bessd gRPC server to become ready")
+            time.sleep(2)
 
-    def _is_bessctl_executed(self) -> bool:
-        """Check if BESSD_CONFIG_CHECK_FILE_NAME exists.
+    def _is_bessd_grpc_service_ready(self) -> bool:
+        """Checks if bessd grpc service is ready.
 
-        If bessctl configure is executed once this file exists.
+        Examines the output from bessctl to see if it is able to communicate
+        with bessd. This indicates the service is ready to accept configuration
+        commands.
 
         Returns:
             bool:   True/False
         """
-        return path_exists(
-            container=self._bessd_container, path=f"/{BESSCTL_CONFIGURE_EXECUTED_FILE_NAME}"
-        )
+        command = "/opt/bess/bessctl/bessctl show version"
+        try:
+            self._exec_command_in_bessd_workload(
+                command=command,
+                timeout=10,
+            )
+            return True
+        except ExecError as e:
+            logger.info("gRPC Check: %s", e)
+            return False
+
+    def _is_bessd_configured(self) -> bool:
+        """Checks if bessd has been configured.
 
-    def _create_bessctl_executed_validation_file(self, content) -> None:
-        """Create BESSCTL_CONFIGURE_EXECUTED_FILE_NAME.
+        Examines the output from bessctl to show worker. If there is no
+        active worker, bessd is assumed not to be configured.
 
-        This must be created outside of the persistent storage volume so that
-        on container restart, bessd configuration will run again.
+        Returns:
+            bool:   True/False
         """
-        push_file(
-            container=self._bessd_container,
-            path=f"/{BESSCTL_CONFIGURE_EXECUTED_FILE_NAME}",
-            source=content,
+        command = "/opt/bess/bessctl/bessctl show worker"
+        try:
+            (stdout, _) = self._exec_command_in_bessd_workload(
+                command=command,
+                timeout=10,
+            )
+            logger.debug("bessd configured workers: %s", stdout)
+            return True
+        except ExecError as e:
+            logger.info(f"Configuration check: {e}")
```

Comment:
The code style guide disallows f-strings in logging calls.
```suggestion
            logger.info("Configuration check: %s", e)
```